src/Sulu/Bundle/AdminBundle/Resources/js/containers/FieldBlocks/FieldBlocks.js

```diff
@@ -372,17 +368,20 @@ class FieldBlocks extends React.Component<FieldTypeProps<Array<BlockEntry>>> {
     };
 
     handleSettingsSubmit = () => {
-        const {onChange, value = []} = this.props;
+        const {onChange} = this.props;
+        const value = this.value || [];
 
         const {blockSettingsFormStore, blockSettingsOpen} = this;
 
         if (!blockSettingsFormStore || blockSettingsOpen === undefined || !value) {
             return;
         }
 
+        value[blockSettingsOpen] = {...value[blockSettingsOpen], [SETTINGS_KEY]: blockSettingsFormStore.data};
```

Comment:
Added the update via the `setValue` method. But as I understand it, we can change the `onChange` call because now the settings are already updated in the `newValues` and thus we only need to make a new object to avoid passing the same reference with `{...newValues[blockSettingsOpen]}` or am I missing something?

Reply:
Not sure what you are talking about, but I already added another review comment, maybe that answers your question :see_no_evil:

Reply:
Yeah I misunderstood it 🙈 . Your snipped helped a lot.